pyanaconda/modules/payloads/payload/dnf/dnf_manager.py

```diff
@@ -127,77 +123,111 @@ def _base(self):
 
         return self.__base
 
+    @property
+    def _goal(self):
+        """The DNF goal."""
+        if self.__goal is None:
+            self.__goal = libdnf5.base.Goal(self._base)
+
+        return self.__goal
+
+    @property
```

Comment:
Can we just have a single goal, aka the original, and setter for that? 

in this way, we can try once, then set again the goal, and try again?

Reply:
If you mean to replace the `_goal` with `_goal_skip_unavailable` only when the first `_goal` doesn't succeed, then it isn't possible. We need both at the beginning, because both must be populated in the `apply_specs` method. We do it the same way in dnf5 e.g. when resolving modules (see https://github.com/rpm-software-management/dnf5/blob/main/libdnf5/module/module_sack.cpp#L477-L484 where four goals are used).